client/client-lib/src/mint/backup.rs

```diff
@@ -168,19 +168,24 @@ impl MintClient {
     pub async fn download_ecash_backup_from_federation(
         &self,
     ) -> Result<Option<PlaintextEcashBackup>> {
-        if let Some(encrypted) = self
+        let mut responses = self
             .context
             .api
             .download_ecash_backup(&self.get_derived_backup_signing_key().x_only_public_key().0)
-            .await?
-        {
-            Ok(Some(
-                EcashBackup(encrypted.data)
-                    .decrypt_with(&self.get_derived_backup_encryption_key())?,
-            ))
-        } else {
-            Ok(None)
+            .await?;
+
+        // try the newest (highest timestamp) first
+        responses.sort_by_key(|backup| Reverse(backup.timestamp));
+
+        for backup in responses {
+            if let Ok(backup) =
+                EcashBackup(backup.data).decrypt_with(&self.get_derived_backup_encryption_key())
```

Comment:
A malicious guardian can feed you an outdated backup since the encryption does not commit to the timestamp (and is in fact determined by the federation instead of the user as far as I'm aware). We could either commit to a user-chosen timestamp with the authenticating encryption scheme or just try to decrypt all and then use the epoch inside to decide which one to use.

To commit to additional plaintext data we'd replace the `Aad::empty()` with `Aad::from(binary_timestamp_repr)`:
https://github.com/fedimint/fedimint/blob/5eb44b183b0af6257248092b86f10360ad8c3a8f/client/client-lib/src/mint/backup.rs#L55